src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs

```diff
@@ -3378,7 +3345,7 @@ private void RunOrQueueCompletionAction(ITaskCompletionAction completionAction,
             }
         }
 
-        private void LogFinishCompletionNotification()
+        private static void LogFinishCompletionNotification()
```

Comment:
Since you've created a bunch of new logging-related methods, it'd be great to group them all together in the same place in the file.